apps/frontend/core-dapp/e2e/utils/wallet-connect.ts

```diff
@@ -55,9 +56,61 @@ export async function connectMockWallet({
     })
 
     topic = session.topic
+
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    const rpcUrl: string = (session.requiredNamespaces.eip155 as any).rpcMap[
+      arbitrumChainId.toString()
+    ]
+    provider = new ethers.providers.JsonRpcBatchProvider(rpcUrl, arbitrumChainId)
   })
 
-  // TODO handle session_request events to sign transactions
+  const onSessionRequest = async (event: Web3WalletTypes.SessionRequest): Promise<void> => {
+    if (!provider || !topic) return
+
+    const { params, id } = event
+    const { request } = params
+
+    if (request.method === 'eth_signTypedData_v4') {
+      const [, payload] = request.params
+      const { domain, types, message } = JSON.parse(payload)
+
+      delete types.EIP712Domain // Ethers errors if we pass this
+      const signed = await wallet._signTypedData(domain, types, message)
+      const response = { id, result: signed, jsonrpc: '2.0' }
+      await web3wallet.respondSessionRequest({ topic, response })
+    } else if (request.method === 'eth_sendTransaction') {
+      const txParams = request.params[0]
+      const txBody = {
+        chainId: arbitrumChainId,
+        data: txParams.data,
+        from: txParams.from,
+        to: txParams.to,
+        value: txParams.value,
+      }
+
+      const [gas, nonce] = await Promise.all([
+        provider.estimateGas(txBody),
+        provider.getTransactionCount(txBody.from),
+      ])
```

Comment:
We should use `getFeeData` here to get maxFeePerGas and maxPriorityFeePerGas so it doesn't keep failing because of gas. I've tried the test command few times and only had 1 success, the rest were failures because of gas price 

```suggestion
      const [gas, nonce, feeData] = await Promise.all([
        provider.estimateGas(txBody),
        provider.getTransactionCount(txBody.from),
        provider.getFeeData()
      ])
```

Reply:
Let me re-evaluate this. I started by pulling the `gasFeeData` but it resulted in overpaying for transactions by a huge margin, so I don't know what went wrong

Reply:
how much did it over pay for? If it's not that much I don't think it'll be a big deal since we'll probably be using some kind of devnets soon and will have unlimited ETH

Reply:
Added the fee data query and it now works more reliably, thanks! I think in my last attempt I was just unlucky and sent the transaction in a time where arbitrum was very expensive.